### PR TITLE
Add bulk annotation addition api

### PR DESCRIPTION
**Pull request by tobrun**
closes https://github.com/mapbox/mapbox-plugins-android/issues/679
needs some tests before review.

### CODE REVIEW

**Comment by tobrun** [comment]
Tests were added + generation of unit tests + fixing stability issues with instrumentation tests

File: app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/utils/OnMapReadyIdlingResource.java [hunk 1]
```diff
@@ -20,9 +19,10 @@
   public OnMapReadyIdlingResource(Activity activity) {
     new Handler(Looper.getMainLooper()).post(() -> {
       try {
-        Field field = activity.getClass().getDeclaredField("mapView");
-        field.setAccessible(true);
-        mapView = ((MapView) field.get(activity));
+        mapView = activity.findViewById(R.id.mapView);
+        if (mapView == null) {
```
Comment (LukasPaczos) [inline app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/utils/OnMapReadyIdlingResource.java]: I think there should be a loop here, right?

Comment (LukasPaczos) [reply app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/utils/OnMapReadyIdlingResource.java]: The checked IDs are different, marking as resolved.

File: plugin-localization/src/main/java/com/mapbox/mapboxsdk/plugins/localization/LocalizationPlugin.java [hunk 2]
```diff
@@ -181,7 +186,10 @@ public void setCameraToLocaleCountry() {
    * @since 0.1.0
    */
   public void setCameraToLocaleCountry(Locale locale) {
-    setCameraToLocaleCountry(checkMapLocalNonNull(locale));
+    MapLocale mapLocale = MapLocale.getMapLocale(locale);
+    if (mapLocale != null) {
+      setCameraToLocaleCountry(mapLocale);
+    }
```
Comment (LukasPaczos) [inline plugin-localization/src/main/java/com/mapbox/mapboxsdk/plugins/localization/LocalizationPlugin.java]: Worth adding a log as in https://github.com/mapbox/mapbox-plugins-android/pull/689/files#diff-1bf159640ccc4e006c457ee997362bf9R112?

Comment (tobrun) [reply plugin-localization/src/main/java/com/mapbox/mapboxsdk/plugins/localization/LocalizationPlugin.java]: nice catch!

**Pull request merged**